### DISCUSSION TITLE
Dataclasses Issue in Local Deployment

### DISCUSSION THREAD

**Discussion by de-wax-apple** [EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge#72] [Q&A]
Hello, 

I am going through the beginner’s guide notebooks but got stuck in the 03_Products_and_measurements when running the notebook in my local machine.
I had manually installed the datacube and odc packages. But it still doesn’t work due to the dataclasses issue. 
 

Originally, there was an error message about missing a “dataclasses” module. I manually copied and paste the dataclasses.py from my Python 3.8 folder into the datacube env’s Python 3.6 folder (Normally, I use Python 3.8, but the default in datacube env is 3.6; so, I used that instead). That fixed this error.
But then another error appeared, saying there's "invalid syntax":
![image001](https://user-images.githubusercontent.com/84428491/119581878-0b0df280-bdf6-11eb-85d1-4f8e74761160.png)

**Comment by EY-glowglowglow** [comment]
@codeindulgence  @alexgleith  can you please have a look at this? Thanks!

**Comment by nickbutler** [comment]
Hi @de-wax-apple.

This is likely caused by using package versions not compatible with 3.6.
How exactly were the packages installed?

The `dataclasses` module is a core Python module introduced in Python 3.7.
Generally speaking, it's very unlikely that copying core modules between Python versions is going to work.
In this case, the version you have copied uses [new syntax which was introduced in Python 3.8](https://docs.python.org/3/faq/programming.html#what-does-the-slash-in-the-parameter-list-of-a-function-mean), which is the cause of the error being shown.

The `datacube` package can be used in Python 3.8, so I would just use that.

Cheers and best of luck!
Let us know how you get on.

**Reply by nickbutler** [reply]
n.b. Just to confirm, I tested it:

```
Python 3.8.2 (default, Jun  3 2020, 14:13:22)
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import datacube
>>> import pandas as pd
>>> from odc.ui import DcViewer
>>> pd.set_option('display.max_colwidth', 200)
>>> pd.set_option('display.max_rows', None)
```

**Reply by de-wax-apple** [reply]
Hi @codeindulgence 

I installed the datacube by following instructions [here](https://datacube-core.readthedocs.io/en/latest/ops/macosx.html#miniconda) with the miniconda.
Seems that by default it is python 3.6 using that method. 

So, if datacube is installed under python 3.6 is there any easy way to open it under python 3.8 mode? I tried running the code using the 3.8 kernel, it doesn't work.

If re-installation is needed,  can you advise how to install datacube under python 3.8 this time?

Thanks.

**Reply by nickbutler** [reply]
Hi @de-wax-apple,

For the purpose of this challenge, we have documentation to help set up a Cube-in-a-Box environment on [Azure](https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/wiki/1c.-Getting-started:-Local-Environment) or [locally](https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/wiki/1c.-Getting-started:-Local-Environment), so issues with setups outside of this are a little bit out of our support scope.

While the documentation on the ODC page recommends 3.6, I presume that this is simply because that was the latest version at the time of writing, and that there would be no issue just replacing `3.6` with `3.8` or even `3.9` (current latest version).

I haven't used Anaconda myself, when I tested it out as above, I just used `pip install ...`.

Cheers
